### PR TITLE
Add Startup and Configuration

### LINKED ISSUE #2: WrapText configuration
Add AppSetting configuration to WrapText (json format)
example:
```javascript
appSettings: {
  wrapText: [
    { label: 'S1', lineId: '190,35', lines: 3, maxChars: 20 },
    { label: 'U2', lineId: '150,25', lines: 5, maxChars: 15 },
...
  ]
}
```

### LINKED ISSUE #7: Use standard MTX Structure
MTX.Conversion.ZebraWrap.Service
MTX.Conversion.ZebraWrap.Contracts
MTX.Conversion.ZebraWrap.Data

### PR DESCRIPTION
**Pull request by HavermansStef**
- Add start up and Configuration
- Add default configuration

- Closes #2 
- Closes #7

### CODE REVIEW

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Helpers/WrapperHelper.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace Contracts.Helpers
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Helpers/WrapperHelper.cs:1]: Empty file

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Service/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿{
+  "iisSettings": {
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Service/Properties/launchSettings.json:2]: Is not a web api

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Service/Configuration/WrapTextValue.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+namespace Service.Configuration
+{
+    public class WrapTextValue
+    {
+        #region Public Members
+
+        public string Label { get; set; }
```
[the remark this answers is not in the document]
Comment (HavermansStef) [reply Zebra2CSV/MTX.Conversion.ZebraWrap.Service/Configuration/WrapTextValue.cs:7]: Fixed.

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Service/Configuration/WrapTextValue.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        
+        public int Lines { get; set; }
+        
+        public int MaxChars { get; set; }
+
+        #endregion
+
+        #region Constructor
+
+        public WrapTextValue()
```
[the remark this answers is not in the document]
Comment (HavermansStef) [reply Zebra2CSV/MTX.Conversion.ZebraWrap.Service/Configuration/WrapTextValue.cs:19]: Fixed.

File: Zebra2CSV/MTX.Conversion.ZebraWrap.App/appsettings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "WrapTextValues": []
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.App/appsettings.json:2]: Adjust as #12

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Constants/FieldDto.cs [hunk 6]
```diff
@@ -1,4 +1,4 @@
-namespace Data
+namespace Contracts.Constants
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Constants/FieldDto.cs:1]: MTX.Conversion.ZebraWrap.Contracts.Constants

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/MTX.Conversion.ZebraWrap.Contracts.csproj [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
     <PropertyGroup>
         <TargetFramework>netcoreapp3.0</TargetFramework>
+        <RootNamespace>Contracts</RootNamespace>
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/MTX.Conversion.ZebraWrap.Contracts.csproj:5]: Correct Namespace

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Parsers/PlnoParser.cs [hunk 8]
```diff
@@ -36,9 +36,12 @@ public string Content
         public WorkListDto Parse()
         {
             if (!IsValid) throw new FormatException();
+            var headerMatch = new Regex(@"(?:\[HEADERSTART\]#\d+#)(?<Author>.+?)(?<Type>#.+?#)").Match(Content);
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Parsers/PlnoParser.cs:39]: `new Regex(@"(?:\[HEADERSTART\]#\d+#)(?<Author>.+?)(?:#)(?<Type>.+?)(?:#.*))`

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+using System.Linq;
+using Contracts.Constants;
+using MTX.Conversion.ZebraWrap.Data.Interfaces.Services;
+
+namespace MTX.Conversion.ZebraWrap.Data.Services
+{
+    public class WrapService : IWrapService
+    {
+        private static readonly List<string> Separators = new List<string>
+            {@" ", @"/", @"\", @":", @";", @"-", @"_", @"=", @".", @","};
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs:11]: Move to configuration

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+                if (_amountOfFields == 0)
+                {
+                    _amountOfFields = label.Fields.Count;
+                }
+                for (var f = 0; f < _amountOfFields; f++)
+                {
+                    var field = label.Fields[f];
+                    var fieldValue = field.Value;
+                    var addedFields = 0;
+                    var maxId = workList.Labels.First(x => x.Id == label.Id).Fields.Select(x => x.Id).Max();
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs:33]: `label.Fields.Max(x => x.Id);`

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+                    // We start counting backwards from the amount of chars
+                    for (var i = Amount; i > 0; i--)
+                    {
+                        // If the length of the field is shorter than the amount of lines per field, put it in the first and fill the rest with empty vals
+                        if (fieldValue.Length < Amount || i == 1)
+                        {
+                            workList.Labels.First(x => x.Id == label.Id).Fields.Add(new FieldDto
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs:41]: `label.FIelds.Add`

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+                        // If no separator is found, continue the loop and ignore the code below
+                        if (!Separators.Contains(fieldValue[i].ToString()))
+                        {
+                            continue;
+                        }
+
+                        addedFields++;
+                        // If it is found, add it to the fields
+                        workList.Labels.First(x => x.Id == label.Id).Fields.Add(new FieldDto
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs:69]: `label`

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+                }
+                for (var f = 0; f < _amountOfFields; f++)
+                {
+                    var field = label.Fields[f];
+                    var fieldValue = field.Value;
+                    var addedFields = 0;
+                    var maxId = workList.Labels.First(x => x.Id == label.Id).Fields.Select(x => x.Id).Max();
+
+                    // We start counting backwards from the amount of chars
+                    for (var i = Amount; i > 0; i--)
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Data/Services/WrapService.cs:36]: Change amount to position

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Tests/App/ConversionServiceTests.cs [hunk 14]
```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
```
Comment (BartVanHoutte-ADING) [inline Zebra2CSV/MTX.Conversion.ZebraWrap.Tests/App/ConversionServiceTests.cs:1]: Rename test to ConfigurationProviderTests

File: Zebra2CSV/MTX.Conversion.ZebraWrap.App/appsettings.json (hunk 5 above)
Comment (HavermansStef) [reply Zebra2CSV/MTX.Conversion.ZebraWrap.App/appsettings.json:2]: Oki

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/MTX.Conversion.ZebraWrap.Contracts.csproj (hunk 7 above)
Comment (HavermansStef) [reply Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/MTX.Conversion.ZebraWrap.Contracts.csproj:5]: Fixed.

File: Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Constants/FieldDto.cs (hunk 6 above)
Comment (HavermansStef) [reply Zebra2CSV/MTX.Conversion.ZebraWrap.Contracts/Constants/FieldDto.cs:1]: Fixed.

**Pull request merged**